fairlearn/grid_search.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""
+This module implements a 1d grid search for binary classification with a single
+binary protected attribute.
+
+This is a simplification of the full exponentiated gradient search
+and produces a sequence of models, each of which corresponds to a
+different accuracy/fairness trade-off
+"""
+
+import copy
+import numpy as np
+import pandas as pd
+from collections import namedtuple
+
+def _generate_p0_p1(Y):
+    """ Function to compute p0 and p1 for the given
+    set of labels Y
+
+    Assume we have already checked that Y is binary with
+    values 0 and 1
+    """
+    unique, counts = np.unique(Y, return_counts=True)
+
+    if len(unique) != 2:
+        raise RuntimeError("Supplied Y not binary")
+
+    if not np.array_equal(unique, [0, 1]):
+        raise RuntimeError("Supplied Y labels not 0 or 1")
+
+    p0 = counts[0] / len(Y)
+    p1 = counts[1] / len(Y)
+
+    return p0, p1
+
+def _generate_weights(Y, A, L, p_ratio):
+    W = []
+
+    for y, a in zip(Y, A):
+        w = 1e128
+        if a==0:
+            w = 2*y - 1 - L * p_ratio
+        else:
+            w = 2*y - 1 + L
+
+        W.append(w)
+
+    return np.array(W)
+
+def classification_binary_protected_1d(learner, X, Y, A, Ls=None, num_Ls = 11):
+    """Function to generate a 1d set of models for a classification problem with
```

Comment:
a 1D set of models --> you mean a list?

Reply:
Indeed - I was being a bit fast and loose in my terminology